source/deployment/mobile-app-deployment.rst

```diff
@@ -0,0 +1,571 @@
+
+**********************************
+Configuring Mattermost to Use HPNS
+**********************************
+
+Configuring your Mattermost Server to use the Mattermost HPNS is a single configuration item. This is covered in our `Hosted Push Notification Service documentation <https://docs.mattermost.com/mobile/mobile-hpns.html>`_. 
+
+Next your users would need to install the mobile application on their device. If desired, you can further configure security capabilities using an EMM provider.
+
+**********************************
+Updating via the Public Store Apps
+**********************************
+
+While not part of your initial mobile deployment, you should consider a strategy for updating when new versions of the Mattermost Mobile applications are available. Simultaneously, you should check any compatibility requirements for the mobile apps and the Mattermost server. 
+
+It is often easier to upgrade the mobile apps. However, not all provided updates are compatible with all previous versions of the Mattermost server. 
+
+    **Note:** Only updating the mobile apps, or updating the mobile apps before the Mattermost server, can result in incompatibility.
+
+----
+
+########################################
+Mobile Deployment via Custom Mobile Apps
+########################################
+
+Choosing this model means you’ve decided not to use the mobile applications Mattermost has made available via the public app stores. 
+
+This also means you will need to maintain these applications. Maintenance includes rebuilding and incorporating feature and/or security updates. Otherwise, your applications will not match the functionality of our publicly available applications, and could be incompatible with future versions of the Mattermost server.
+
+Finally, this choice also means you cannot use the hosted version of the Mattermost Push Notification Service (HPNS).
+
+A great first step for this deployment model involves `reviewing the documentation <https://developers.mattermost.com/contribute/mobile/>`_. Your development team should run through the requirements for building the applications. This will help understand the overall impact when choosing this option. This documentation provides guidance on building, compiling, and signing. It also includes information for customizing the apps.
+
+We’ve provided some recommended sections below.
+
+- Developer Setup
+- Build Your Own App
+- Push Notifications
+
+*********************************
+Distribution through an app store
+*********************************
+
+Once you have built your own apps you will need to distribute them. There are two options:
+
+Option 1 - Enterprise App Store via EMM Provider (Most Common): 
+
+- This is the most common way for customers to distribute their apps.
+- Once added to your own "Enterprise App Store", user can download.
+
+Option 2 - Apple App Store & Google Play (not recommended):
```

Comment:
Why not recommended?

Reply:
For the audience of this guide we intended to provide the most efficient path to mobile deployment. Given that, the recommendation followed to the more direct, efficient, or quicker path.

Reply:
Ok so how is it not efficient for me as a small/medium/large company to use the available options out there and use them?

I may not have an Enterprise app store or an EMM and want to deploy the customized apps to my personal and simply publish it in the Apple App Store or Google Play under a specific name like `OurCompanyChat`. The only pain point may be going through the Apple review process that can take a couple of days or so.

In fact, if I do not already have an Enterprise app store or an EMM provider in place it may be even more painful to set it up just to deploy the mattermost app (regardless of the associated costs).

Maybe just remove the `(not recommended)`  and add to option one (most common and recommended)

Reply:
I'm ok with this suggestion. For the sake of those scanning we should add (less common) in place of (not recommended) to option 2 as well.